src/java/org/apache/cassandra/service/ActiveRepairService.java

```diff
@@ -499,6 +499,18 @@ public void handleMessage(InetAddress endpoint, RepairMessage message)
         }
     }
 
+    public boolean isRepairing(UUID cfId)
+    {
+        for (ParentRepairSession session : parentRepairSessions.values())
```

Comment:
ParentRepairSession created when a coordinator starts a repair: https://github.com/palantir/cassandra/blob/31dd7de91aaa0bb4db8af82e46fd068817afd5e7/src/java/org/apache/cassandra/repair/RepairRunnable.java#L219 and when a receiving node gets a repair message: https://github.com/palantir/cassandra/blob/31dd7de91aaa0bb4db8af82e46fd068817afd5e7/src/java/org/apache/cassandra/repair/RepairMessageVerbHandler.java#L84-L89. But while the session does seem to be cleaned up on the coordinator's end: [removeParentRepairSession(parentRepairSession)](https://github.com/palantir/cassandra/blob/df3fdf21a944ad1420375421cae24db3f250bbb5/src/java/org/apache/cassandra/service/ActiveRepairService.java#L450), I am not seeing it being cleaned up on the receiver end?

Reply:
We said that Anticompaction does not run because we do not run incremental repairs. This is true, however, the anticompaction messages are still sent, and each node handles the message with the following logic

https://github.com/palantir/cassandra/blob/714d4f92cf4ccd17620d77b5ae885139398a8867/src/java/org/apache/cassandra/service/ActiveRepairService.java#L436-L452

Reply:
The call stack is 
- `RepairRunnable`
- `ActiveRepairService.instance.finishParentSession`
- for-each neighbor: `AnticompactionTask#run` (sends the message)
- (jump to remote)
- `RepairMessageVerbHandler#doVerb() ANTICOMPACTION_REQUEST`
- `ActiveRepairService.instance.doAntiCompaction`
- removeParentRepairSession

Reply:
Niice, sgtm!